aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h

```diff
@@ -190,5 +211,160 @@ struct conv_param_t {
       ukernel_type = pytorch_qnnp_ukernel_type_conv;
     }
   }
+
+  /**
+   * @brief Computes the output dimensions given a 2D input.
+   */
+  std::array<size_t, 2> compute_output_dims(
+      std::array<size_t, 2> input_dims) const {
+    std::array<size_t, 2> output_dims;
+    output_dims[0] = compute_output_dimension(input_dims[0],  // width
+                                              pad[1] + pad[3],
+                                              /*adjustment=*/0,
+                                              kernel_dims[0],
+                                              dilation[0],
+                                              subsampling_dims[0],
+                                              /*transpose=*/false);
+    output_dims[1] = compute_output_dimension(input_dims[1],  // height
+                                              pad[0] + pad[2],
+                                              /*adjustment=*/0,
+                                              kernel_dims[1],
+                                              dilation[1],
+                                              subsampling_dims[1],
+                                              /*transpose=*/false);
+    return output_dims;
+  }
+};
+
+struct deconv_param_t {
+  const std::array<uint32_t, 2> kernel_dims; // kernel width, kernel height
+  const std::array<uint32_t, 2> stride_dims; // stride width, height
+  const std::array<uint32_t, 2> dilation_dims; // dilation width, height
+  const std::array<uint32_t, 4> padding_dims; // input padding top, left, bottom, right
+  const std::array<uint32_t, 2> adjustment_dims; // output adjustment
+
+  const uint32_t groups;
+  const size_t input_channels;
+  const size_t output_channels;
+  const uint8_t kernel_zero_point;
+  const float kernel_scale;
+  const uint8_t output_min;
+  const uint8_t output_max;
+
+  // The following are derived parameters
+  enum pytorch_qnnp_ukernel_type ukernel_type; // kernel type based on input params
+  size_t group_input_channels;
+  size_t group_output_channels;
+
+  /**
+   * @brief Constructor for initializing the deconvolution parameters.
+   */
+  deconv_param_t(
+      const std::array<uint32_t, 2> kernel,
+      const std::array<uint32_t, 2> stride,
+      const std::array<uint32_t, 2> dil,
+      const std::array<uint32_t, 4> pd,
+      const std::array<uint32_t, 2> adj,
```

Comment:
We can do better names. Perhaps trail names with _ to avoid confusion?

Reply:
These names are coming directly from the original QNNPACK conv_param_t naming. Besides, what can this be confused with? I don't mind changing the var names, but that calls for refactor of the qnnpack as a whole.

Reply:
Just a more readability argument. Looking at pd does not tell me anything right away, or `dil`. I then have to look a few lines down. I was commenting from a practice perspective. I am okish if you wanna leave it like that but changing here doesn't have to be coupled with conv_params_t. That one can be done in a separate PR. I leave it upto you.